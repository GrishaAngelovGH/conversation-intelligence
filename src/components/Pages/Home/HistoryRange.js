import Button from 'react-bootstrap/Button'
import ButtonGroup from 'react-bootstrap/ButtonGroup'

const HistoryRange = () => {
  const ranges = ['Last 24h', 'Last 7 days', 'Last 30 days', 'This Month', 'Custom']

  return (
    <ButtonGroup>
      {
        ranges.map((v, i) => (
          <Button key={i} variant='outline-primary'>{v}</Button>
        ))
      }
    </ButtonGroup>
  )
}

export default HistoryRange