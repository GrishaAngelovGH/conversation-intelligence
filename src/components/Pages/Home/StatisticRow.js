import PropTypes from 'prop-types'

const StatisticRow = ({ data }) => (
  <div className="row border border-2 rounded">
    <div className="col">
      <div className="fw-bold">{new Date().toDateString()}</div>
    </div>
    <div className="col">
      <div className="fw-bold">Q2, 48 days left</div>
    </div>
    {
      data.map((v, i) => (
        <div key={i} className="col">
          <div className="fw-bold">{v.type}: {v.value}</div>
        </div>
      ))
    }
  </div>
)

StatisticRow.propTypes = {
  data: PropTypes.array.isRequired
}

export default StatisticRow