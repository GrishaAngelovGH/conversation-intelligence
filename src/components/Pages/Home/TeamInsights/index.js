import TeamInsights from "./TeamInsights"

export default TeamInsights