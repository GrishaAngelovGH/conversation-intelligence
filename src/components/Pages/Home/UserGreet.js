const UserGreet = ({ username }) => (
  <h3 className="fw-bold">Good morning, {username}</h3>
)

export default UserGreet