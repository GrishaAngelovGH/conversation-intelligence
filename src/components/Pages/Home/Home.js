import UserGreet from './UserGreet'
import HistoryRange from './HistoryRange'
import StatisticRow from './StatisticRow'
import StatisticCards from './StatisticCards'

const Home = () => {

  const statisticData = [
    { type: 'Actual Rev.', value: '$300k' },
    { type: 'Open Rev.', value: '$150k' },
    { type: 'Deals-Won', value: '$280k' },
    { type: 'Win-Rate', value: '63%' },
    { type: 'Avg. Deal Size', value: '58%' }
  ]

  return (
    <div className="row g-0 justify-content-center text-center">
      <div className="col-md-11">

        <div className="row g-0 p-3">
          <div className="col-md-6">
            <UserGreet username="James Smith" />
          </div>
          <div className="col-md-6">
            <HistoryRange />
          </div>
        </div>

        <StatisticRow data={statisticData} />

        <StatisticCards />
      </div>
    </div>
  )
}

export default Home