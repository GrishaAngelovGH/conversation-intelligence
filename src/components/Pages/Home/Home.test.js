import { render } from '@testing-library/react'

import Home from './Home'

test('should render component', () => {
  const container = render(<Home />)

  expect(container).toMatchSnapshot()
})