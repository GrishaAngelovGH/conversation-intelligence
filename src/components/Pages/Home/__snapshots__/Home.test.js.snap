// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row g-0 justify-content-center text-center"
      >
        <div
          class="col-md-11"
        >
          <div
            class="row g-0 p-3"
          >
            <div
              class="col-md-6"
            >
              <h3
                class="fw-bold"
              >
                Good morning, 
                James Smith
              </h3>
            </div>
            <div
              class="col-md-6"
            >
              <div
                class="btn-group"
                role="group"
              >
                <button
                  class="btn btn-outline-primary"
                  type="button"
                >
                  Last 24h
                </button>
                <button
                  class="btn btn-outline-primary"
                  type="button"
                >
                  Last 7 days
                </button>
                <button
                  class="btn btn-outline-primary"
                  type="button"
                >
                  Last 30 days
                </button>
                <button
                  class="btn btn-outline-primary"
                  type="button"
                >
                  This Month
                </button>
                <button
                  class="btn btn-outline-primary"
                  type="button"
                >
                  Custom
                </button>
              </div>
            </div>
          </div>
          <div
            class="row border border-2 rounded"
          >
            <div
              class="col"
            >
              <div
                class="fw-bold"
              >
                Mon Oct 17 2022
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="fw-bold"
              >
                Q2, 48 days left
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="fw-bold"
              >
                Actual Rev.
                : 
                $300k
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="fw-bold"
              >
                Open Rev.
                : 
                $150k
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="fw-bold"
              >
                Deals-Won
                : 
                $280k
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="fw-bold"
              >
                Win-Rate
                : 
                63%
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="fw-bold"
              >
                Avg. Deal Size
                : 
                58%
              </div>
            </div>
          </div>
          <div
            class="row mt-3"
          >
            <div
              class="col-md-12"
            >
              <h3
                class="fw-bold"
              >
                What are your customers talking about?
              </h3>
              <p>
                Understand what's happening on sales calls, so you can coachyour sellers better.
              </p>
              <div
                class="row border border-2 rounded shadow"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row align-items-center"
                  >
                    <div
                      class="col-md-6"
                    >
                      <canvas
                        height="0"
                        role="img"
                        style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
                        width="0"
                      />
                    </div>
                    <div
                      class="col-md-6"
                    >
                      <div>
                        <h4>
                          Tracked keywords 
                          <span
                            class="fw-bold"
                          >
                            3D Printing
                          </span>
                           and 
                          <span
                            class="fw-bold"
                          >
                            Manufacturing
                          </span>
                           are trending upward across sales calls
                        </h4>
                        <p
                          class="text-muted"
                        >
                          Tracked keywords can provide insights into what your customers are talking about, which could surface sales opportunities.
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row g-0 justify-content-center text-center"
    >
      <div
        class="col-md-11"
      >
        <div
          class="row g-0 p-3"
        >
          <div
            class="col-md-6"
          >
            <h3
              class="fw-bold"
            >
              Good morning, 
              James Smith
            </h3>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="btn-group"
              role="group"
            >
              <button
                class="btn btn-outline-primary"
                type="button"
              >
                Last 24h
              </button>
              <button
                class="btn btn-outline-primary"
                type="button"
              >
                Last 7 days
              </button>
              <button
                class="btn btn-outline-primary"
                type="button"
              >
                Last 30 days
              </button>
              <button
                class="btn btn-outline-primary"
                type="button"
              >
                This Month
              </button>
              <button
                class="btn btn-outline-primary"
                type="button"
              >
                Custom
              </button>
            </div>
          </div>
        </div>
        <div
          class="row border border-2 rounded"
        >
          <div
            class="col"
          >
            <div
              class="fw-bold"
            >
              Mon Oct 17 2022
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="fw-bold"
            >
              Q2, 48 days left
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="fw-bold"
            >
              Actual Rev.
              : 
              $300k
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="fw-bold"
            >
              Open Rev.
              : 
              $150k
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="fw-bold"
            >
              Deals-Won
              : 
              $280k
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="fw-bold"
            >
              Win-Rate
              : 
              63%
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="fw-bold"
            >
              Avg. Deal Size
              : 
              58%
            </div>
          </div>
        </div>
        <div
          class="row mt-3"
        >
          <div
            class="col-md-12"
          >
            <h3
              class="fw-bold"
            >
              What are your customers talking about?
            </h3>
            <p>
              Understand what's happening on sales calls, so you can coachyour sellers better.
            </p>
            <div
              class="row border border-2 rounded shadow"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-md-6"
                  >
                    <canvas
                      height="0"
                      role="img"
                      style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
                      width="0"
                    />
                  </div>
                  <div
                    class="col-md-6"
                  >
                    <div>
                      <h4>
                        Tracked keywords 
                        <span
                          class="fw-bold"
                        >
                          3D Printing
                        </span>
                         and 
                        <span
                          class="fw-bold"
                        >
                          Manufacturing
                        </span>
                         are trending upward across sales calls
                      </h4>
                      <p
                        class="text-muted"
                      >
                        Tracked keywords can provide insights into what your customers are talking about, which could surface sales opportunities.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
